src/utils/aepp-wallet-communication/wallet-detector.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2022 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * Wallet Detector
+ *
+ * This is the complement to {@link module:@aeternity/aepp-sdk/es/utils}.
+ * @module @aeternity/aepp-sdk/es/utils/aepp-wallet-communication/wallet-detector
+ * @export WalletDetector
+ */
+import BrowserWindowMessageConnection from './connection/browser-window-message'
+import { MESSAGE_DIRECTION, METHODS } from './schema'
+import { UnsupportedPlatformError } from '../errors'
+import WalletConnection from '../aepp-wallet-communication/connection'
+import { WalletInfo } from './rpc/aepp-rpc'
+
+interface Wallets {
+  [key: string]: WalletInfo
+}
+
+/**
+ * WalletDetector
+ * @alias module:@aeternity/aepp-sdk/es/utils/aepp-wallet-communication/wallet-detector
+ * @param params - Initializer object
+ * @returns WalletDetector
+ */
+export default class WalletDetector {
+  connection: WalletConnection
+  wallets: Wallets
+
+  constructor ({ connection }: { connection: WalletConnection }) {
```

Comment:
I don't see why we need to accept a `connection` object here, it can be an independent param.
i see this a lot, Am I missing anything?
```suggestion
  constructor (connection?: WalletConnection) {
```